bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/handler/UpnpRendererHandler.java

```diff
@@ -0,0 +1,866 @@
+                    "Communication lost with " + thing.getLabel());
+        }
+        super.onStatusChanged(status);
+    }
+
+    @Override
+    public void onValueReceived(@Nullable String variable, @Nullable String value, @Nullable String service) {
+        if (logger.isDebugEnabled() && !("AbsTime".equals(variable) || "RelCount".equals(variable)
+                || "RelTime".equals(variable) || "AbsCount".equals(variable) || "Track".equals(variable)
+                || "TrackDuration".equals(variable))) {
+            // don't log all variables received when updating the track position every second
+            logger.debug("Upnp device {} received variable {} with value {} from service {}", thing.getLabel(),
+                    variable, value, service);
+        }
+        if (variable == null) {
+            return;
+        }
+
+        switch (variable) {
+            case "CurrentMute":
+                if (!((value == null) || (value.isEmpty()))) {
+                    soundMute = OnOffType.from(Boolean.parseBoolean(value));
+                    updateState(MUTE, soundMute);
+                }
+                break;
+            case "CurrentVolume":
+                if (!((value == null) || (value.isEmpty()))) {
+                    soundVolume = PercentType.valueOf(value);
+                    updateState(VOLUME, soundVolume);
+                }
+                break;
+            case "Sink":
+                if (!((value == null) || (value.isEmpty()))) {
+                    updateProtocolInfo(value);
+                }
+                break;
+            case "LastChange":
+                // pre-process some variables, eg XML processing
+                if (!((value == null) || value.isEmpty())) {
+                    if ("AVTransport".equals(service) && "LastChange".equals(variable)) {
+                        Map<String, String> parsedValues = UpnpXMLParser.getAVTransportFromXML(value);
+                        for (Map.Entry<String, String> entrySet : parsedValues.entrySet()) {
+                            // Update the transport state after the update of the media information
+                            // to not break the notification mechanism
+                            if (!"TransportState".equals(entrySet.getKey())) {
+                                onValueReceived(entrySet.getKey(), entrySet.getValue(), service);
+                            }
+                            if ("AVTransportURI".equals(entrySet.getKey())) {
+                                onValueReceived("CurrentTrackURI", entrySet.getValue(), service);
+                            } else if ("AVTransportURIMetaData".equals(entrySet.getKey())) {
+                                onValueReceived("CurrentTrackMetaData", entrySet.getValue(), service);
+                            }
+                        }
+                        if (parsedValues.get("TransportState") != null) {
```

Comment:
This will eliminate the compiler warning.
```suggestion
                        if (parsedValues.containsKey("TransportState")) {
```